### PR TITLE
Implement CloudLinux 7 upgrades

### PR DESCRIPTION
**Pull request by sloanebernstein**
Add preliminary support to handle the ELevate process using CloudLinux as the start and end points.

By submitting pull requests to this repo, I agree to the Contributor License Agreement which can be found at: https://github.com/cpanel/elevate/blob/main/docs/cPanel-CLA.pdf

### CODE REVIEW

File: elevate-cpanel [hunk 1]
```diff
@@ -5818,7 +5859,7 @@ sub _do_leapp_upgrade ($self) {
 
     my $ok = eval {
         local $ENV{LEAPP_OVL_SIZE} = 3000;
-        $self->ssystem_and_die( { keep_env => 1 }, qw{/usr/bin/leapp upgrade} );
+        $self->ssystem_and_die( { keep_env => 1 }, qw{/usr/bin/leapp upgrade --nowarn} );
```
Comment (troglodyne) [inline elevate-cpanel:5862]: I am somewhat curious why this was added. Were the warnings just unhelpful noise?

**Comment by troglodyne** [comment]
Do we need to update the docs as well for this?

**Comment by sloanebernstein** [comment]
I'll add a subtask for the docs.

File: lib/Elevate/Blockers/Repositories.pm [hunk 2]
```diff
@@ -83,6 +87,8 @@ use constant VETTED_YUM_REPO => qw{
   elasticsearch
   elevate
   elevate-source
+  elevate-testing
```
Comment (toddr) [inline lib/Elevate/Blockers/Repositories.pm:90]: no thank you.

File: script/elevate-cpanel.PL [hunk 3]
```diff
@@ -886,12 +905,22 @@ sub run_stage_3 ($self) {
...
                         $self->ssystem_and_die(qw{/usr/bin/yum install -y http://repo.almalinux.org/elevate/elevate-release-latest-el7.noarch.rpm});
+                        $self->ssystem_and_die(qw{/usr/bin/curl -o /etc/yum.repos.d/ELevate-testing.repo https://repo.almalinux.org/elevate/testing/elevate-testing.repo});
                     }
 
-                    my $leap_data_pkg = $self->upgrade_to_rocky() ? 'leapp-data-rocky' : 'leapp-data-almalinux';
+                    my $leapp_data_pkg;
+                    if ( $self->upgrade_to_rocky() ) {
+                        $leapp_data_pkg = 'leapp-data-rocky';
+                    }
+                    if ( $self->upgrade_to_cloudlinux() ) {
```
Comment (atoomic) [inline script/elevate-cpanel.PL:915]: should not this be an `elsif` ?

**Comment by toddr** [comment]
We will not merge this pull request until a fresh install of CloudLinux7 can upgrade to CloudLinux 8 off the production LEAPP repos.

**Comment by toddr** [comment]
https://docs.cloudlinux.com/shared/elevate/#elevate-scenario-cloudlinux-7-with-no-panel-or-a-custom-panel

**Comment by toddr** [comment]
this will be replaced by a PR from @cPholloway

**Pull request closed without merging**